src/identity.js

```diff
@@ -439,90 +439,79 @@ export class Identity extends EventEmitter {
         if (this._hasSessionInProgress) {
             return this._hasSessionInProgress;
         }
-        const promiseFn = async (resolve, reject) => {
-            const postProcess = (sessionData) => {
-                if (sessionData.error) {
-                    this.emit('error', sessionData.error);
-                    return reject(new SDKError('HasSession failed', sessionData.error));
-                }
-                this._maybeSetVarnishCookie(sessionData);
-                this._emitSessionEvent(this._session, sessionData);
-            };
-
-            if (typeof autologin !== 'boolean') {
-                const [type, value] = inspect(autologin);
-                return reject(new SDKError(`Parameter 'autologin' must be boolean, was: "${type}:${value}"`));
+        if (typeof autologin !== 'boolean') {
+            const [type, value] = inspect(autologin);
+            return Promise.reject(new SDKError(`Parameter 'autologin' must be boolean, was: "${type}:${value}"`));
+        }
+        const _postProcess = (sessionData) => {
+            if (sessionData.error) {
+                throw new SDKError('HasSession failed', sessionData.error);
             }
-
-            try {
-                if (this._enableSessionCaching) {
-                    // Try to resolve from cache (it has a TTL)
-                    const cachedData = this.cache.get(HAS_SESSION_CACHE_KEY);
-                    if (cachedData) {
-                        postProcess(cachedData);
-                        return resolve(cachedData);
+            this._maybeSetVarnishCookie(sessionData);
+            this._emitSessionEvent(this._session, sessionData);
+        };
+        const _getSession = async () => {
+            let sessionData = null;
+            if (this._enableSessionCaching) {
+                // Try to resolve from cache (it has a TTL)
+                sessionData = this.cache.get(HAS_SESSION_CACHE_KEY);
```

Comment:
One issue here, since you are not returning value here, code https://github.com/schibsted/account-sdk-browser/pull/124/files#diff-eaef817870a2a9f52ca5c3938e630207R494 is being executed. As a result cache never expires, cache is updated with new `expiresOn` value.
Can we add `_postProcess` here and return a value?

Reply:
That’s a nice catch! Would be neat with a test for this as well to ensure cache is only updated when we expect it to

Reply:
Fixed the code. Haven't added a test for this case though